pennylane/circuit_drawer/grid.py

```diff
@@ -0,0 +1,193 @@
+    """
+
+    def __init__(self, raw_grid=None):
+        if raw_grid is None:
+            # Let initialization pend until first data is entered
+            self.raw_grid = None
+        else:
+            self.raw_grid = np.array(raw_grid, dtype=object)
+
+    def insert_layer(self, idx, layer):
+        """Insert a layer into the Grid at the specified index.
+
+        If the Grid is empty, the index is ignored and the layer inserted.
+
+        Args:
+            idx (int): Index at which to insert the new layer
+            layer (list): Layer that will be inserted
+        """
+        if self.raw_grid is not None:
+            self.raw_grid = np.insert(self.raw_grid, idx, np.array([layer], dtype=object), axis=1)
+        else:
+            self.raw_grid = np.array([layer], dtype=object).T
+
+        return self
+
+    def append_layer(self, layer):
+        """Append a layer to the Grid.
+
+        Args:
+            layer (list): Layer that will be appended
+        """
+        if self.raw_grid is not None:
+            self.raw_grid = np.append(self.raw_grid, np.array([layer], dtype=object).T, axis=1)
+        else:
+            self.raw_grid = np.array([layer], dtype=object).T
+
+        return self
+
+    def replace_layer(self, idx, layer):
+        """Replace a layer in the Grid at the specified index.
+
+        Args:
+            idx (int): Index of the layer to be replaced
+            layer (list): Layer that replaces the old layer
+        """
+        if self.raw_grid is not None:
+            self.raw_grid[:, idx] = np.array(layer, dtype=object)
+        else:
+            raise AttributeError("Can't replace layer. The Grid has not yet been initialized.")
+
+        return self
+
+    def insert_wire(self, idx, wire):
+        """Insert a wire into the Grid at the specified index.
+
+        If the Grid is empty, the index is ignored and the wire inserted.
+
+        Args:
+            idx (int): Index at which to insert the new wire
+            wire (list): Wire that will be inserted
+        """
+        if self.raw_grid is not None:
+            self.raw_grid = np.insert(self.raw_grid, idx, np.array([wire], dtype=object), axis=0)
+        else:
+            self.raw_grid = np.array([wire], dtype=object)
+
+        return self
+
+    def append_wire(self, wire):
+        """Append a wire to the Grid.
+
+        Args:
+            wire (list): Wire that will be appended
+        """
+        if self.raw_grid is not None:
+            self.raw_grid = np.append(self.raw_grid, np.array([wire], dtype=object), axis=0)
+        else:
+            self.raw_grid = np.array([wire], dtype=object)
+
+        return self
+
+    @property
+    def num_layers(self):
+        """Number of layers in the Grid.
+
+        Returns:
+            int: Number of layers in the Grid
+        """
+        if self.raw_grid is not None:
+            return self.raw_grid.shape[1]
+
+        return 0
```

Comment:
Could this become a one-liner?
```python
return 0 if self.raw_grid is None else self.raw_grid.shape[1]
```

Reply:
It could, but I'd leave it as is. It makes no big difference and the current version is easier to read